Spell Updates

jakeswork opened the pull request:
Great work on this. I've made some changes to the base repo over my time using it and thought I'd share.

The major difference is opting to use spell ids rather than spell names. When using the strings, it just wasn't working consistently (only half of the spells would show). Swapping to IDs seems to have fixed this

This PR also includes
- A more comprehensive spell list - some stuff was missing that wotlk players would be interested in seeing
- A better solution to displaying icons that don't have durations. Previously it would show the cooldown overlay and made the icon just look dark - the solution I opted for was to only render the cooldown overlay should there be a cooldown to show
- _minor_: separated spells into classes, makes it easier to navigate/spot missing important spells

I'd recommend pulling down the original and my fork and playing around with it to really see the benefits.

I plan on adding more to this also - this PR is so that (should you want my updates) the changes aren't as drastic in future PRs.

jakeswork committed:
Initial update - get ALL spells working by using IDs rather than names

Takes the same approach the original BigDebuffs project did. Strings are unreliable and is the reason for many spells not working as expected on the wotlk client

jakeswork committed:
Organise spells into relevant class categories

Also added shadow dance, re-prioritised fear ward and removed tiger's fury (not relevant enough of a buff to warrant an icon)

jakeswork committed:
Removing tiger's fury

jakeswork committed:
reshuffling buffs around

jakeswork committed:
Add mana tide, sham rage, freeze and tremor totem

jakeswork committed:
Add earthgrab

jakeswork committed:
fix drink

jakeswork committed:
Add freezing arrow effect

jakeswork committed:
Adding hunter pet skills

jakeswork committed:
Add shield of the templar silence

jakeswork committed:
Add hunter pet intervene, change warr intervene to defensive, fix test mode

jakeswork committed:
Fix IBF

jakeswork committed:
Add stoneclaw stun

jakeswork committed:
fix divine shield

jakeswork committed:
fix intercept

jakeswork committed:
add impact

jakeswork committed:
fix impact

jakeswork committed:
Add turn evil and holy wrath

jakeswork committed:
fix wind shear duration

jakeswork committed:
add divine sacrifice

jakeswork committed:
Only show the cooldown timer if there is a cooldown to show

Previously it showed the overlay completely over spells that didnt have a timer, e.g. grounding totem

jakeswork committed:
swap to cs id, add imp cs

jakeswork committed:
Add imporved kick silence

jakeswork committed:
Add blanket spell lock and update id

jakeswork committed:
move cyclone to immunities

jakeswork committed:
Add starfall

jakeswork committed:
Add hunter pet disarm

jakeswork committed:
Add web (hunter pet)

jakeswork committed:
Add hunter disarm

jakeswork committed:
Add shield bash, update pummel to use id

jakeswork committed:
Add arcane torrent(s)

jakeswork committed:
Add shield block

jakeswork committed:
Update anti magic zone (may still not work)

jakeswork committed:
Add cower (hunter pet defensive)

jakeswork committed:
Add hunter turtle pet defensive

jakeswork committed:
Add hunter pet interrupt

jakeswork committed:
Add huddle

jakeswork committed:
Add felguard intercept

jakeswork committed:
Add metamorphosis

jakeswork committed:
Add demon charge

jakeswork committed:
Add hymn of hope

jakeswork committed:
Add bestial wrath, update the beast within and deterrence to immunities

jakeswork committed:
Add retal

jakeswork committed:
Add taste for blood and UA

jakeswork committed:
Add death wish

jakeswork committed:
Add concussion blow

jakeswork committed:
Add revenge stun

jakeswork committed:
Add scare beast

jakeswork committed:
Add rapid fire

jakeswork committed:
Add counterattack

jakeswork committed:
Add entrapment 1/3, 2/3, 3/3

jakeswork committed:
Add fiery payback

jakeswork committed:
Add divine plea

jakeswork committed:
Add seal of justice stun

jakeswork committed:
Add cold blood

jakeswork committed:
add shaman andr duid natures swiftness

jakeswork committed:
Add psychic horror horrify and disarm

jakeswork committed:
add mage water

jakeswork committed:
Updating final strings to ids - keep it consistent

jakeswork committed:
Add natures grasp and tidy up spells that had multiple ids whilst testing

jakeswork committed:
Basic creds, updated readme to be a little more github noob friendly

jakeswork committed:
minor aesthetic tweak - cooldown countainer was a liiiiil too small on arena frames

jakeswork committed:
fix for spells with no duration

https://gyazo.com/82fbd3a0f3c3e1ac75fe15417aed6dd0

ManneN1 commented:
Looks like good work. I tested it briefly. What other changes do you mean when talking about potentially adding more things?

jakeswork commented:
> Looks like good work. I tested it briefly. What other changes do you mean when talking about potentially adding more things?

Cheers. The only piece of "new" functionality that a friend of mine suggested was the ability to toggle spells on individual frames

At the moment, we can only set an individual spell or category of spells to appear on `Unit Frames` - ideally we could have
| Display |  |
| -------| - |
| [✓] Player Frame | [   ] Arena Frames |
| [   ] Target Frame |  [✓] Focus Frame |

jakeswork committed:
Add silencing shot

jakeswork committed:
Add enh wolves bash and sprint, remove tremor (useless, theres no aura), fix intercept

jakeswork committed:
Adding absorbs - setting them all to "other" so they're not too intrusive

jakeswork committed:
add gag order silence

jakeswork committed:
Add burning determination, remove combustion

jakeswork committed:
Update Intimidating Shout (there are two auras)

jakeswork committed:
Fix elemental mastery

Was showing the haste buff rather than instant cast buff

jakeswork committed:
Update shield bash duration

jakeswork committed:
Fix strang

jakeswork committed:
Add stoneclaw totem absorb

jakeswork committed:
Add r1 shadowfury

jakeswork committed:
Add fingers of frost

jakeswork committed:
Add druid shapeshift forms

jakeswork committed:
add the art of war

jakeswork committed:
Adds warrior stances. A bit hacky, but if we want this feature this is the only way I can see it possible

jakeswork committed:
Add roots from NG, move lastKnownStance logic

ManneN1 committed:
Update BigDebuffs.lua

Changing credits

ManneN1 committed:
Update BigDebuffs.toc

Adding credit to readme instead.

ManneN1 committed:
Update Options.lua

ManneN1 committed:
Update README.md

ManneN1 committed:
Update BigDebuffs.lua

We want to keep all the aura selection logic all in one place, including stances (and interrupts, but that's already there).

ManneN1 committed:
Minor fix
